frame/ibc/ibc-rpc/src/lib.rs

```diff
@@ -1354,41 +1378,58 @@ where
 		})
 	}
 
-	fn query_newly_created_clients(
+	fn query_newly_created_client(
 		&self,
 		block_hash: Block::Hash,
-	) -> Result<Vec<IdentifiedClientState>> {
+		ext_hash: Block::Hash,
+	) -> Result<IdentifiedClientState> {
 		let api = self.client.runtime_api();
 		let at = BlockId::Hash(block_hash);
-		let events = api
-			.block_events(&at)
-			.map_err(|_| runtime_error_into_rpc_error("[ibc_rpc]: failed to read block events"))?;
-
-		let mut identified_clients = vec![];
-		for e in events {
-			match e {
-				IbcEvent::CreateClient { client_id, .. } => {
-					let result: ibc_primitives::QueryClientStateResponse = api
-						.client_state(&at, client_id.clone())
-						.ok()
-						.flatten()
-						.ok_or_else(|| runtime_error_into_rpc_error("client state to exist"))?;
-
-					let client_state = AnyClientState::decode_vec(&result.client_state)
-						.map_err(|_| runtime_error_into_rpc_error("client state to be valid"))?;
-					let client_state = IdentifiedClientState {
-						client_id: String::from_utf8(client_id).map_err(|_| {
-							runtime_error_into_rpc_error("client id should be valid utf8")
-						})?,
-						client_state: Some(client_state.into()),
-					};
-					identified_clients.push(client_state)
-				},
-				_ => continue,
-			}
-		}
+		let block = self.client.block(&at).ok().flatten().ok_or_else(|| {
+			runtime_error_into_rpc_error("[ibc_rpc]: failed to find block with provided hash")
+		})?;
+		let extrinsics = block.block.extrinsics();
+		let (ext_index, ..) = extrinsics
+			.iter()
+			.enumerate()
```

Comment:
Do you need enumerate?

Reply:
i guess that's how you get the index?

Reply:
Yeah, I don't think there's any other way to get the index when searching through the items of a `Vec`